languages/ruby/exercises/concept/floating-point-numbers/.docs/introduction.md

```diff
@@ -0,0 +1,13 @@
+A floating-point number is a number with zero or more digits behind the decimal separator. Examples are `4.0`, `0.1`, `3.14`, `-6.4` `16.984025` and `1024.0`.
+
+In Ruby, floating-point numbers are implemented through the [Float](https://ruby-doc.org/core-2.7.0/Float.html) class.
+
+In this exercise you may also want to use a loop. There are several ways to write loops in Ruby, one of them is the `while` loop:
```

Comment:
So this is the interesting part of this exercise. I'd nearly always use `(1...5).each ...` in this scenario in reality. I suspect we should introduce that either in this exercise or in a prerequisite. This might be a good exercise to introduce it in, but it will make it a bit more of a complex exercise. Because we've already introduced constants and conditionals though, I think it's a reasonable thing to do.

So let's have a think about how we introduce ranges and enumeration.

Reply:
That's just an example of a while loop. The actual method the student is supposed to write solves a different problem than the one in the example. I'm not sure if a `Range` is the best option for solving the problem. The problem:

### 3. Calculate the years before reaching the desired balance

Implement the `SavingsAccount#years_before_desired_balance` method to calculate the minimum number of years required to reach the desired balance:

```ruby
SavingsAccount.years_before_desired_balance(200.75, 214.88)
#=> 14
```

Reply:
I see. Yes. Sorry. OK. I'm good with this then. Thanks :)